sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -93,22 +97,16 @@ object AnalysisContext {
  * Provides a logical query plan analyzer, which translates [[UnresolvedAttribute]]s and
  * [[UnresolvedRelation]]s into fully typed objects using information in a [[SessionCatalog]].
  */
-class Analyzer(
+abstract class Analyzer(
```

Comment:
Why not continue to pass a lookup function in some paths and default the analyzer to use one that throws `CatalogNotFoundException`? That would avoid a lot of these changes.

```scala
class Analyzer(..., catalogLookup: String => CatalogPlugin) {
  def this(...) = {
    this(..., _ => throw new CatalogNotFoundException(...)
  }
}

```

Reply:
Reverted commit `Make Analyzer abstract and add TestAnalyzer` to minimize changes to test classes.

If you feel strongly, I can add the 4th parameter `override val lookupCatalog: String => CatalogPlugin` to Analyzer constructor.

Reply:
Sounds good to me.